### PR TITLE
Extending endpoint tests

### PR DESCRIPTION
**Pull request by EricDavisX**
Feature test work for Agent & Endpoint integration
In support of https://github.com/elastic/e2e-testing/issues/336

Replaces PR https://github.com/elastic/e2e-testing/pull/351 which will be closed in favor of this.
## What does this PR do?

This PR re-uses existing steps from the Agent side and includes tests written there to include when Endpoint is deployed (included in Policy) to ensure Endpoint behaves as expected as well.

It does not include any new go implementation, just feature file work.

## Why is it important?
This is important because Agent controls Endpoint's calls to start, stop, deploy, and general communication.  Also, we found a bug recently that could have been caught if we had these tests (that we were planning on doing and had not gotten around to).

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
~~- [ ] I have run the Unit tests for the CLI, and they are passing locally~~
~~- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally~~
~~- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)~~

### CODE REVIEW

**Review by mdelapenya — APPROVED** [review]
This is awesome!

image

Let's wait for the CI to pass before merging

Thanks!

**Comment by EricDavisX** [comment]
i'll rebase when prior pr merges and get it passed linting. :)

File: e2e/_suites/fleet/features/agent_endpoint_integration.feature
Comment (mdelapenya) [inline e2e/_suites/fleet/features/agent_endpoint_integration.feature:65]: EOF in Gherkin files is checked by precommit:

```suggestion
    And the "elastic-endpoint" process is in the "started" state on the host

```

**Comment by mdelapenya** [comment]
> i'll rebase when prior pr merges and get it passed linting. :)

TW, you have a `make -C e2e lint` from root dir to give you some hints

**Comment by mdelapenya** [comment]
@EricDavisX to avoid the `AvoidScripting` message in Given clauses, we could create a new step as a shortcut.

Instead of:
```gherkin
Given an Endpoint is successfully deployed with a "centos" Agent using "tar" installer
    And the agent is un-enrolled
    And the "elastic-agent" process is "stopped" on the host
  When the agent is re-enrolled on the host
    And the "elastic-agent" process is "started" on the host
```
we could have:
```gherkin
Given the "elastic-agent" process is stopped in an already un-enrolled "centos" Agent using "tar" installer with an Endpoint deployed
When the "elastic-agent" process is "started" on a re-enrolled host
```
And we can omit the Centos+Tar part to simplify the sentence:
```gherkin
Given the "elastic-agent" process is stopped in an already un-enrolled Agent with an Endpoint deployed
When the "elastic-agent" process is "started" on a re-enrolled host
```

wdyt?

**Comment by mdelapenya** [comment]
Hey @EricDavisX, the errors reported in this PR are expected to be fixed in #475

**Comment by EricDavisX** [comment]
> @EricDavisX to avoid the `AvoidScripting` message in Given clauses, we could create a new step as a shortcut.
> 
> And we can omit the Centos+Tar part to simplify the sentence: 
> ```
> Given the "elastic-agent" process is stopped in an already un-enrolled Agent with an Endpoint deployed
> When the "elastic-agent" process is started on a re-enrolled host
> ```
> wdyt?

Hi, thanks for the thinking on it.  I am hoping that we can at some near point try to tackle adding more OSes into the fray of our testing.  I don't know if removing the parameterization is helpful in that regards, tho I get that we are not using more params yet.   I'd like to slack or zoom about the linting's anger and understand more first.  I do generally like having the 'simplification' summary steps tho, as you had listed - it makes quicker reading for the same file and if it is more compact it is great.  Maybe doing the middle step, leaving in tar + centos options is the way to go.

**Comment by EricDavisX** [comment]
one last try on pulling in master, lol, and using latest artifacts then I'll dig deeper on why Endpoint is failing here, may need to pull in the Endpoint team

**Comment by EricDavisX** [comment]
we found a potential reason the test wasn't working - so pulling in master one more time after noted merge.

**Comment by EricDavisX** [comment]
I don't know what's going on, but I think the docker auth update recently from IT messed up my local branch / fork and I can't sync to upstream, and I was reducing the scope of changes anyhow - so I'm just manually copying the work from here to https://github.com/elastic/e2e-testing/pull/947

**Pull request closed without merging**
